processors/base/writers/location.py

```diff
@@ -70,3 +76,37 @@ def write_location(conn, location, source_id, trial_id=None):
             'created' if create else 'updated', name)
 
     return object['id']
+
+#----helper function to get the canonical name of the locations
+def normalize_location(location):
+    location_name = location.lower().strip().replace(".", "")
+    try:
+        canonical_name = countries.get(location_name).name
+        logger.debug('Location - %s: %s normalized', location_name, canonical_name)
+    except KeyError:
+        canonical_name = location
+        # This part is for looking for matching cases (suggestion), uses the 
+        # fuzzy logic to get the country name who get 80% of match rate.
+        with open(os.path.join(os.path.dirname(__file__),'countries.csv'), 'r') as f:
```

Comment:
Extract the CSV path to a variable and add a whitespace between `os.path.join` parameters